http/webhook.go

```diff
@@ -0,0 +1,87 @@
+package http
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/grafana/alerting/receivers"
+)
+
+type ForkedSender struct {
+	cli receivers.WebhookSender
+}
+
+func NewForkedSender(cli receivers.WebhookSender) *ForkedSender {
+	return &ForkedSender{cli: cli}
+}
+
+func (f ForkedSender) SendWebhook(ctx context.Context, cmd *receivers.SendWebhookSettings) error {
+	if cmd.HTTPMethod != "GET" {
+		return f.cli.SendWebhook(ctx, cmd)
+	}
+
+	request, err := http.NewRequestWithContext(ctx, cmd.HTTPMethod, cmd.URL, nil)
+	if err != nil {
+		return err
+	}
+	_, err = url.Parse(cmd.URL)
+	if err != nil {
+		// Should not be possible - NewRequestWithContext should also err if the URL is bad.
+		return err
+	}
+
+	request.Header.Set("User-Agent", "Grafana")
+
+	if cmd.User != "" && cmd.Password != "" {
+		request.Header.Set("Authorization", GetBasicAuthHeader(cmd.User, cmd.Password))
```

Comment:
There's no need to create a function for this, Go already has one in its stdlib.

```suggestion
		request.SetBasicAuth(cmd.User, cmd.Password)
```